### PR TITLE
Query for channel modes AFTER room mappings are created

### LINKED ISSUE #362: Dynamic mapping to a +i channel does not create invite-only room
Hi,

I tried to join a room corresponding to a channel with the +i mode, and the created room is accessible to "Anyone who knows the room's link, apart from guests".

Reproduction:
- on the IRC network, join a new channel
- set mode +i
- join it from matrix

Version: 0.7.1 (a2b025683176924fcc5abb8dbb05b75572fc6b33)
Network: irc.iiens.net (UnrealIRCd)

Here is the `debug.log` file when trying with a channel name `#ohlol`: [ohlol.txt](https://github.com/matrix-org/matrix-appservice-irc/files/791112/ohlol.txt).

### PR DESCRIPTION
**Pull request by Kegsay**
Fixes #362

### CODE REVIEW

File: lib/bridge/MatrixHandler.js [hunk 1]
```diff
@@ -425,6 +425,14 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
             yield this.ircBridge.getStore().storeRoom(
                 ircRoom, mxRoom, 'join'
             );
+            // /mode the channel AFTER we have created the mapping so we process
+            // +s and +i correctly.
+            this.ircBridge.publicitySyncer.initModeForChannel(ircServer, ircRoom.channel).catch(
```
Comment (lukebarnard1) [inline lib/bridge/MatrixHandler.js]: could use `ircChannel` instead of `ircRoom.channel`

File: lib/bridge/MatrixHandler.js [hunk 2]
```diff
@@ -425,6 +425,14 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
...
+            (err) => {
+                log.error(
+                    `Could not init mode for channel ${ircRoom.channel} on ${ircServer.domain}`
```
Comment (lukebarnard1) [inline lib/bridge/MatrixHandler.js]: as above

File: lib/bridge/MatrixHandler.js (hunk 1 above)
Comment (Kegsay) [inline lib/bridge/MatrixHandler.js]: Done.

File: lib/bridge/MatrixHandler.js (hunk 2 above)
Comment (Kegsay) [inline lib/bridge/MatrixHandler.js]: Done.

**Pull request merged**